### PR TITLE
Stabilize the HyphenTest.network unit test.

### PR DESCRIPTION
**Pull request by stoklund**
This unit test was failing intermittently because it depended too closely on
the numbers delivered by nextRand(). The intermittent behavior can be observed
with:

    tests/JITTest --gtest_filter=HyphenTest.network --gtest_repeat=25

Make this test less sensitive by:

1. Use a deterministically seeded RNG for shuffling the training data.
2. Increate the training mini batch size 10 -> 25.
3. Double the number of training iterations.

With these changes, the test still runs in ~600 ms.

### CODE REVIEW

File: tests/unittests/HyphenTest.cpp [hunk 1]
```diff
@@ -334,10 +333,15 @@ TEST(HyphenTest, network) {
 
   // Randomly shuffle the training data.
```
Comment (nadavrot) [inline tests/unittests/HyphenTest.cpp]: At the moment this is the only unit test that uses it's own mt19937 random number generator. All other tests use nextRand().  Let's assume that all future tests would mimic this test. Do we want people to instantiate their own random number generator? Or maybe we want to encourage the tests to be more stable by changing the hyper parameters? Another alternative would be to adda reset function to the internal random number generator. 

What do you think?

File: tests/unittests/HyphenTest.cpp [hunk 2]
```diff
@@ -33,6 +33,7 @@
 
 #include <cassert>
 #include <cctype>
+#include <random>
```
Comment (rdzhabarov) [inline tests/unittests/HyphenTest.cpp]: i think <cassert> and <cctype> can be removed.

File: tests/unittests/HyphenTest.cpp (hunk 1 above)
Comment (rdzhabarov) [reply tests/unittests/HyphenTest.cpp]: I think in general tests should be resilient to random gen, but if it's not possible (hard to achieve) explicit setting seed might be an option.

Comment (stoklund) [reply tests/unittests/HyphenTest.cpp]: Agreed about using stable tests. This network is clearly somewhat wobbly. I've increased the batch size further to 50, but there just isn't a lot of training data in this test case.

We could encourage tests to use the standard library, or we could provide our own pseudo-random number facility, either is fine by me. I do think the global state in `nextRand()` is problematic, though. Let's discuss more over in #1066.

I'm going to land this as is to unblock people from adding more JIT tests. When we reach a consensus on #1066, I can update this test to match.

File: tests/unittests/HyphenTest.cpp (hunk 2 above)
Comment (stoklund) [reply tests/unittests/HyphenTest.cpp]: Thanks. Fixed
